notebook/static/notebook/js/notebook.js

```diff
@@ -1958,26 +1958,48 @@ define(function (require) {
         dialog.modal(options.dialog);
         return promise;
     };
-    
+
     /**
-     * Execute or render cell outputs and go into command mode.
+     * Execute cells corresponding to the given indices.
+     *
+     * @param {list} indices - indices of the cells to execute
      */
-    Notebook.prototype.execute_cell = function () {
-        // mode = shift, ctrl, alt
-        var cell = this.get_selected_cell();
-        
-        cell.execute();
+    Notebook.prototype.execute_cells = function (indices) {
+        if (indices.length === 0) {
+            return;
+        }
+
         this.command_mode();
         this.set_dirty(true);
+
+        var cell;
+        for (var i = 0; i < indices.length; i++) {
+            this.select(indices[i]);
+            cell = this.get_cell(indices[i]);
+            cell.execute();
+        }
+    };
+
+    /**
+     * Execute or render cell outputs and go into command mode.
+     */
+    Notebook.prototype.execute_cell = function () {
```

Comment:
Yeah. The reason I didn't change this name was because it is used in other locations already (the `execute_cell` function is not one I added, just modified). If you think it is going to make things clearer though I can change the name here and elsewhere.